### PR TITLE
bpo-44524: Don't modify MRO when inheriting from typing.Annotated

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
https://bugs.python.org/issue44524

### CODE REVIEW

File: Lib/typing.py
Comment (serhiy-storchaka) [inline Lib/typing.py:1578]: It restores the past issue: `__name__` returning non-string.

I have a patch with other solution of this issue. After merging it your PR would not cause problems.

Comment (serhiy-storchaka) [reply Lib/typing.py:1578]: See #27861.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by Fidget-Spinner** [comment]
Wait don't merge yet, I need to update some things.

**Comment by Fidget-Spinner** [comment]
Interestingly, `Annotated[Any, 'foo'].__name__` now returns `Any`, because it sets origin to `Any`.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**
